components/supervisor/pkg/ports/ports.go

```diff
@@ -5,6 +5,7 @@
 package ports
 
 import (
+	"bufio"
```

Comment:
Could you elaborate on changes in this file please?

Reply:
Generally it looks good. I'm not entirely sure why global port should go away, and we need to fix tests.